### ISSUE TITLE
Batch Dimension squeeze

### ISSUE THREAD

**Issue by StinkyBenji** [sergivalverde/nicMSlesions#16]
Hey Sergi, I was training some dataset using nicMSlesions, however, the following error occurred.

![Screenshot from 2020-11-19 09-03-08](https://user-images.githubusercontent.com/55496957/99807088-3fe77700-2b3f-11eb-85c7-46649a3f7629.png)

So I guess there is  one batch shape for a subject is (1,2,11,11,11), and in base.py, test_scan,  line 551: y_pred = model['net'].predict(np.squeeze(batch),options['batch_size'])
you used np.squeeze(batch), so the shape becomes (2, 11, 11, 11)? 
So do you think I can simply modify the script by adding a condition after np.squeeze the batch or
sqz_batch = np.squeeze(batch)
if len(sqz_batch.shape) < 5:
sqz_batch = np.expand_dims(sqz_batch, axis = 0)
else....

Looking forward to your reply.
Thanks

**Comment by sergivalverde** [comment]
Hi,

Looks like for this particular image, it exists only one lesion patch [1,2,11,11,11] so it's squeezed. What you suggest may work. Maybe you can delete the np.squeeze in line [551](https://github.com/sergivalverde/nicMSlesions/blob/9aba4b8b569a57f22d429e7728797bb688fc23e9/libs/CNN/base.py#L551)?

Let me know what it does.

s

**Comment by ohhhhhoooooohhh** [comment]
Hey,
ystd I used the condition that I suggested in the base.py and ran the process for a small epoch on a small training set, there was no error occurring anymore. Bcs I wasn't sure what's ur initial intention for using np.squeeze(), thus I didn't want to simply delete it.. Anyway the modified script works just fine :)

**Comment by isabellameds** [comment]
Dear Sergi,
I am having the same problem, but it occurs when I try to infer the segmentation. 
![error_teste_imagem](https://user-images.githubusercontent.com/70056500/110362836-901c1380-8020-11eb-99b7-0c0afa18eec0.png)
What's the best way to solve it?

Kind regards,
Isabella
